invite/src/invite_bot.ts

```diff
@@ -16,12 +16,24 @@
 
 import {Octokit} from '@octokit/rest';
 
+import {dbConnect} from './db';
 import {GitHub} from './github';
 import {ILogger, Invite, InviteAction} from './types';
 import {InvitationRecord} from './invitation_record';
 
-// TODO: Enable after filling in implementations.
-/* eslint-disable @typescript-eslint/no-unused-vars */
+const INVITE_MACROS: Record<string, InviteAction> = {
+  invite: InviteAction.INVITE,
+  tryassign: InviteAction.INVITE_AND_ASSIGN,
+};
+
+// Regex source: https://github.com/shinnn/github-username-regex
+const USERNAME_PATTERN: string = '[a-z\\d](?:[a-z\\d]|-(?=[a-z\\d])){0,38}';
+const MACRO_PATTERN: string = `/(${Object.keys(INVITE_MACROS).join('|')})`;
+const FULL_MACRO_REGEX: RegExp = new RegExp(
+  // (?<!\\s) ensures the macro is not preceded by a non-space character.
+  `(?<!\\S)${MACRO_PATTERN} @${USERNAME_PATTERN}`,
```

Comment:
You can use `\b` instead of `(?<!\S)`. e.g., this'll catch "I think Ryan can help here./tryassign @rcebulko" - which is much more likely an unintentional missing space than it is a non-intention to invoke the bot

(if you still prefer to use the negative lookbehind thingy, fix the comment that says `\s` instead of `\S`)

Reply:
`\b` actually matches anything that precedes `/`, since it is considered a word break. I fiddled with this for a while lol. I could explicitly add punctiation I suppose, but my concern was that there's also the case where suppose we're commenting on this repo and we said "I'm planning to move ./*.js to ./invite @danielrozenberg" . ie, would rather make it not trigger on a typo than accidentally trigger on some random comment. WDYT?

Reply:
Fair